src/main/java/org/craftercms/engine/service/impl/SiteItemServiceImpl.java

```diff
@@ -139,7 +139,7 @@ public SiteItem getSiteItem(String url, ItemProcessor processor, Predicate<Item>
 
         Item item = storeService.findItem(context.getContext(), null, url, processor);
         if (item != null && (predicate == null || predicate.evaluate(item))) {
-            return createItemWrapper(item);
+            return createItemWrapper(new Item(item));
```

Comment:
Should we move this to the `createItemWrapper` method instead?

Reply:
We can do that if we change `createItemWrapper` to a generic function such as:
```
protected <T> SiteItem createItemWrapper(T item) {
    if (item instanceof Item) {
        return new DefaultSiteItem(new Item((Item) item), modelFieldConverter, sortComparator);
    } else if (item instanceof Tree) {
        return new DefaultSiteItem(new Tree((Tree) item), modelFieldConverter, sortComparator);
    } else {
        throw new IllegalArgumentException("Unsupported type: " + item.getClass());
    }
}
```

I don't have a strong opinion as of current. Creating `new Item/Tree` on certain methods is also fine if we would like to have some methods to return the same item (for performance) in the future.

Reply:
Thanks, now I see. I think it is better the way it is now.